java/fory-core/src/main/java/org/apache/fory/resolver/XtypeResolver.java

```diff
@@ -232,16 +228,126 @@ private void register(
+   * classinfo.
+   */
+  public <T> void setSerializer(
+      Class<T> cls, Serializer<T> serializer, String namespace, String typeName, int xtypeId) {
+    addSerializer(cls, serializer, namespace, typeName, xtypeId);
+  }
+
+  /** Ass serializer for specified class. */
+  private void addSerializer(
+      Class<?> type, Serializer<?> serializer, String namespace, String typeName, int xtypeId) {
+    Preconditions.checkNotNull(serializer);
+    ClassInfo classInfo = classInfoMap.get(type);
+
+    if (classInfo == null) {
+      classInfo = newClassInfo(type, serializer, namespace, typeName, (short) xtypeId);
+      classInfoMap.put(type, classInfo);
+    }
+
+    // 2. Set `Serializer` for `ClassInfo`.
+    classInfo.serializer = serializer;
+  }
+
+  private Class<? extends Serializer> getObjectSerializerClass(
+      Class<?> cls,
+      boolean shareMeta,
+      boolean codegen,
+      JITContext.SerializerJITCallback<Class<? extends Serializer>> callback) {
+    if (codegen) {
+      if (classCtx.contains(cls)) {
+        // avoid potential recursive call for seq codec generation.
+        return CodegenSerializer.LazyInitBeanSerializer.class;
+      } else {
+        try {
+          classCtx.add(cls);
+          Class<? extends Serializer> sc;
+          switch (fory.getCompatibleMode()) {
+            case SCHEMA_CONSISTENT:
+              sc =
+                  fory.getJITContext()
+                      .registerSerializerJITCallback(
+                          () -> ObjectSerializer.class,
+                          () -> loadCodegenSerializer(fory, cls),
+                          callback);
+              return sc;
+            case COMPATIBLE:
+              // If share class meta, compatible serializer won't be necessary, class
+              // definition will be sent to peer to create serializer for deserialization.
+              sc =
+                  fory.getJITContext()
+                      .registerSerializerJITCallback(
+                          () -> shareMeta ? ObjectSerializer.class : CompatibleSerializer.class,
+                          () ->
+                              shareMeta
+                                  ? loadCodegenSerializer(fory, cls)
+                                  : loadCompatibleCodegenSerializer(fory, cls),
+                          callback);
+              return sc;
+            default:
+              throw new UnsupportedOperationException(
+                  String.format("Unsupported mode %s", fory.getCompatibleMode()));
+          }
+        } finally {
+          classCtx.remove(cls);
+        }
+      }
+    } else {
+      if (codegen) {
+        LOG.info("Object of type {} can't be serialized by jit", cls);
+      }
+      switch (fory.getCompatibleMode()) {
+        case SCHEMA_CONSISTENT:
+          return ObjectSerializer.class;
+        case COMPATIBLE:
+          return shareMeta ? ObjectSerializer.class : CompatibleSerializer.class;
```

Comment:
CompatibleSerializer.class is not used for xlang serialization

Reply:
shareMeta ?  MetaSharedSerializer.class : ObjectSerializer.class
is this correctly？